### PR TITLE
chore(config): propagate ValidatedConfig through telemetry directory

### PR DESCRIPTION
**Pull request by rwaskiewicz**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Unit tests (`npm test`) were run locally and passed
- [x] E2E Tests (`npm run test.karma.prod`) were run locally and passed
- [x] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

`Config` is used in a few remaining functions in the `cli/telemetry` directory. This leads to an
 inaccurate count of `strictNullChecks` violations
GitHub Issue Number: N/A

## What is the new behavior?

There are three functions that were updated as a part of this PR.

See each commit message for the details, although they're all largely the same, with https://github.com/ionic-team/stencil/pull/3484/commits/775aeee02f57e825b97dece5e3856ca1d7534104 probably being the more interesting of the three

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

Unit tests continue to pass

## Other information

We move the needle forward by _2_ _whole_ violations here.  That's .07949125596% less than we had 🎉 🙃

### CODE REVIEW

File: src/cli/telemetry/telemetry.ts [hunk 1]
```diff
@@ -92,7 +101,15 @@ export function isUsingYarn(sys: d.CompilerSystem) {
...
-export async function getActiveTargets(config: d.Config): Promise<string[]> {
+/**
+ * Build a list of the different types of output targets used in a Stencil configuration.
+ *
+ * Duplicate entries will not be returned from the list
+ *
+ * @param config the configuration used by the Stencil project
+ * @returns a unique list of output target types found in the Stencil configuration
+ */
+export async function getActiveTargets(config: d.ValidatedConfig): Promise<string[]> {
```
Comment (alicewriteswrongs) [inline src/cli/telemetry/telemetry.ts:112]: we don't have to change this necessarily, but any idea why this function is `async`?

File: src/cli/telemetry/telemetry.ts [hunk 2]
```diff
@@ -190,16 +207,16 @@ const CONFIG_PROPS_TO_DELETE: ReadonlyArray<keyof d.Config> = ['sys', 'logger',
...
+  const anonymizedConfig: d.Config = { ...config };
 
   for (const prop of CONFIG_PROPS_TO_ANONYMIZE) {
     if (anonymizedConfig[prop] !== undefined) {
       anonymizedConfig[prop] = 'omitted';
     }
   }
 
-  anonymizedConfig.outputTargets = (config.outputTargets ?? []).map((target) => {
+  anonymizedConfig.outputTargets = config.outputTargets.map((target) => {
```
Comment (alicewriteswrongs) [inline src/cli/telemetry/telemetry.ts:219]: 🎉🎉🎉

**Review by alicewriteswrongs — APPROVED** [review]
looks great! 👍

File: src/cli/telemetry/telemetry.ts (hunk 1 above)
Comment (rwaskiewicz) [reply src/cli/telemetry/telemetry.ts:112]: i'm gonna guess it was a copy paste error when it was orignally added.  No time like the present though I suppose, should be easy enough to fix

**Review by alicewriteswrongs — APPROVED** [review]
lgtm! 👍

**Pull request merged**